mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ClientState.scala

```diff
@@ -249,7 +256,9 @@ import scala.util.{Failure, Success}
             disconnect(context, data.connect.connectFlags, data.remote, data)
           case (context, ReceiveConnAckTimeout) =>
             disconnect(context, data.connect.connectFlags, data.remote, data)
-          case (_, e) if data.stash.size < data.settings.maxClientConnectionStashSize =>
+          case (context, ConnectionLost) =>
+            disconnect(context, data.connect.connectFlags, data.remote, data)
+          case (_, e) =>
```

Comment:
We now handle a connection being lost within the client side. This had been forgotten.